src/Plugin/Condition/AdoType.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+namespace Drupal\format_strawberryfield\Plugin\Condition;
+
+use Drupal\Core\Condition\ConditionPluginBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\webform\Entity\WebformOptions;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides the 'ADO Type' condition.
+ *
+ * @Condition(
+ *   id = "ado_type_condition",
+ *   label = @Translation("ADO Type"),
+ *   context_definitions = {
+ *     "node" = @ContextDefinition("entity:node", label = @Translation("node"))
+ *   }
+ * )
+ */
+class AdoType extends ConditionPluginBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * Creates a new AdoType instance.
+   *
+   * @param array $configuration
+   *   The plugin configuration.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $type_options = [];
+    foreach(['schema_org_creative_works', 'schema_org_cw_collections'] as $webform_option_list_id) {
+      $webform_option_list = WebformOptions::load($webform_option_list_id);
+      if (!empty($webform_option_list)) {
+        $type_options = array_merge($type_options, $webform_option_list->getOptions());
+      }
+    }
+    asort($type_options);
+    $form['ado_types'] = [
+      '#title' => $this->pluginDefinition['label'],
+      '#type' => 'checkboxes',
+      '#options' => $type_options,
+      '#default_value' => $this->configuration['ado_types'],
+    ];
+    return parent::buildConfigurationForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+    $this->configuration['ado_types'] = array_filter($form_state->getValue('ado_types'));
+    parent::submitConfigurationForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function evaluate() {
+    // Returns true if no ado types are selected and negate option is disabled.
+    if (empty($this->configuration['ado_types']) && !$this->isNegated()) {
+      return TRUE;
+    }
+    /** @var \Drupal\node\Entity\Node $node */
+    $node = $this->getContextValue('node');
+
+    // Get the ADO type from the entity.
+    if ($node->hasField('field_descriptive_metadata')) {
```

Comment:
@patdunlavey this here needs to change. There are archipelagos in the wild (e.g the ones using EADs) that have different field/names. The way we do this in almost every spot is using this helper methods.

An example:
```
 if ($sbf_fields = \Drupal::service('strawberryfield.utility')->bearsStrawberryfield($entity)) 
```

But a fuller example is here: (The View Mode resolver)


https://github.com/esmero/format_strawberryfield/blob/8f4a77212d4443212df8c72d980fca8fbb00d686/src/ViewModeResolver.php#L95-L110

Which also leads to:
- Should we be aligned with how the View Mode resolver works? Using all "type" key values found in the JSON (independent of the hierarchy).

```

Reply:
Good suggestion. I'll just "borrow" the view mode resolver code to get the type(s).

Reply:
@patdunlavey thanks so much